File: Python/Arrays/D7_StockBuyAndSell2.py
def maximum_profit(prices):
    return sum(
            prices[i] - prices[i - 1]
            for i in range(1, len(prices))
            if prices[i] > prices[i - 1]
    )


prices = [100, 180, 260, 310, 40, 535, 695]
print(maximum_profit(prices))  

File: Java/Arrays/D7_StockBuyAndSell2.java
package Arrays;

public class D7_StockBuyAndSell2 {
    public static void main(String[] args) {
        int[] prices = {100, 180, 260, 310, 40, 535, 695};
        System.out.println(maximumProfit(prices));
    }
    public static int maximumProfit(int[] prices) {
        int n = prices.length;
        int profit = 0;
        for (int i = 1; i < n; i++) {
            if (prices[i] > prices[i - 1]) {
                profit += prices[i] - prices[i - 1];
            }
        }
        return profit;
    }
}
